### DISCUSSION TITLE
how to limit table of contents depth per post basis

### DISCUSSION THREAD

**Discussion by zoomequipd** [mmistakes/minimal-mistakes#3158] [Q&A]
In looking for possible solutions I found two major "clues"
1) https://github.com/mmistakes/minimal-mistakes/issues/2158
2) https://github.com/mmistakes/minimal-mistakes/issues/108

while using toc_levels as described in Issue #108 seems to be deprecated, overriding the entire toc.html as mentioned in #2158 would apply to every post.

My thought was to add a toc_h_min and toc_h_max to the frontmatter and then support using them within the single.html as follows
```
        {% if page.toc %}
          {% assign h_min = page.toc_h_min | default: 1 %}
          {% assign h_max = page.toc_h_max | default: 6 %}
          <aside class="sidebar__right {% if page.toc_sticky %}sticky{% endif %}">
            <nav class="toc">
              <header><h4 class="nav__title"><i class="fas fa-{{ page.toc_icon | default: 'file-alt' }}"></i> {{ page.toc_label | default: site.data.ui-text[site.locale].toc_label | default: "On this page" }}</h4></header>
              {% include toc.html sanitize=true html=content h_min=h_min h_max=h_min class="toc__menu" skip_no_ids=true %}
            </nav>
          </aside>
        {% endif %}
``` 

Is there a better way to do this?  I'd rather not have to overwrite the single.html layout if possible.

**Comment by iBug** [comment]
I guess overriding `single.html` is the best you can get. Don't be afraid to customize, this theme is not updated as often and as breaking.
